src/main/java/org/lightcouch/internal/URIBuilder.java

```diff
@@ -115,10 +115,37 @@ public URIBuilder pathToEncode(String path) {
         return this;
     }
 
-    public URIBuilder query(String name, Object value) {
-        if (name != null && value != null)
-            this.qParams.addParam(name, value.toString());
+    /**
+     * Add the given {@code name} and {@code value} to the query parameters or update
+     * the existing query parameter matching {@code name} with the new {@code value}.
+     * @param name The name to add/update.
+     * @param value The value to add/update.
+     * @param update set to true to update an existing query parameter matching {@code name},
+     *               or false to add a new one. Note that if this is true and there is no
+     *               parameter matching {@code name}, the parameter will be added.
+     * @return The updated {@link URIBuilder} object.
+     */
+    public URIBuilder query(String name, Object value, boolean update) {
+        if (name != null && value != null) {
+            if (update) {
+                this.qParams.updateOrAdd(name, value.toString());
```

Comment:
I'm pretty certain Cloudant/Couch doesn't use the same name for query parameters, thoughts @mikerhodes